### DISCUSSION TITLE
Support custom loader or plugin with Rust

### DISCUSSION THREAD

**Discussion by richerfu** [web-infra-dev/rspack#3221] [Ideas]
### What problem does this feature solve?

We know that build time not only depends on the AST operations but also on other operations. If we use `JS/TS` to write `custom loaders/plugins`, I think it may cause a decrease in compilation speed. 

### What does the proposed API of configuration look like?

Is it possible to provide a `rust crate` to write corresponding loaders/plugins based on Rust, load the corresponding `crates` natively, or build corresponding executable files through `napi-rs` for loading and use?

**Comment by richerfu** [comment]
refer: https://www.rspack.dev/guide/introduction.html#compared-with-webpack
<img width="812" alt="image" src="https://github.com/web-infra-dev/rspack/assets/32590310/048f6719-474f-4d81-863a-45005094fdd9">

**Comment by richerfu** [comment]
I got it with F&Q: https://www.rspack.dev/misc/FAQ.html#do-custom-plugins-and-custom-loaders-need-to-be-developed-using-rust?

**Comment by hyf0** [comment]
We currently have no plan for this feature, but feedbacks are welcome.

**Comment by ununian** [comment]
How about using WebAssembly?
